backend/cmd/headlamp.go

```diff
@@ -437,6 +657,131 @@ func StartHeadlampServer(config *HeadlampConfig) {
 	log.Fatal(http.ListenAndServe(":"+config.port, handler))
 }
 
+func GetFreePort() (int, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
+
+	defer l.Close()
+
+	return l.Addr().(*net.TCPAddr).Port, nil
+}
+
+// nolint: funlen
+func (c *HeadlampConfig) startPortForward(p PortForwardPayload, token string) {
+	ports := []string{fmt.Sprintf(p.Port + ":" + p.TargetPort)}
+	ctxtProxy, ok := c.contextProxies[p.Cluster]
+
+	if !ok {
+		log.Fatalf("cluster %s not found", p.Cluster)
```

Comment:
don't use `Fatalf` return the error and fail the request, in this case even if the port forward is not setup the endpoint will not return an error.

Reply:
fixed